packages/graphql-transformer-core/src/ITransformer.ts

```diff
@@ -0,0 +1,102 @@
+import {
+    DirectiveNode,
+    ObjectTypeDefinitionNode,
+    InterfaceTypeDefinitionNode,
+    FieldDefinitionNode,
+    UnionTypeDefinitionNode,
+    EnumTypeDefinitionNode,
+    ScalarTypeDefinitionNode,
+    InputObjectTypeDefinitionNode,
+    InputValueDefinitionNode,
+    EnumValueDefinitionNode,
+    DirectiveDefinitionNode,
+    TypeDefinitionNode
+} from 'graphql'
+import TransformerContext from './TransformerContext'
+
+export default interface ITransformer {
```

Comment:
This interface simplifies the process of building transformers. Previously, we expected an instance of a class that extended `Transformer`. Moving towards interfaces allows for that use case while also allowing transformers to be written as plain JS objects.